Dockerfile-develop

```diff
@@ -52,13 +52,18 @@ RUN git show > octopus-source-version
 RUN echo "octopus-source-clone-date: $date " >> octopus-source-version
 
 RUN autoreconf -i
-RUN ./configure --enable-mpi --enable-openmp --with-blacs="-lscalapack-openmpi"
+
+# We need to set FCFLAGS_ELPA as the octopus m4 has a bug
+# see https://gitlab.com/octopus-code/octopus/-/issues/900
+RUN export FCFLAGS_ELPA="-I/usr/include -I/usr/include/elpa/modules" && \
+    ./configure --enable-mpi --enable-openmp --with-blacs="-lscalapack-openmpi"
+
 # Which optional dependencies are missing?
 RUN cat config.log | grep WARN > octopus-configlog-warnings
 RUN cat octopus-configlog-warnings
 
 # all in one line to make image smaller
-RUN make && make install && make clean && make distclean
+RUN make -j && make install && make clean && make distclean
```

Comment:
Should we add `-j` in the other file (`Dockerfile`) as well?

Reply:
I wanted to test if the `-j` has any speedup on the GitHub CI, Let me trigger them ( i changed the target branch which is why no CI ran) and then make a decision.

Reply:
[with -j](https://github.com/fangohr/octopus-in-docker/actions/runs/7703658239) the run time is 10 minutes while the [other](https://github.com/fangohr/octopus-in-docker/actions/runs/7703660050) takes 15 minutes, so we can add -j to both branches. ( I was not sure if the runners on GitHub have multiple core that could make use of the `-j`)

Reply:
Also good for local builds.